runtime/hsa-runtime/core/inc/amd_xdna_driver.h

```diff
@@ -44,9 +44,35 @@
 
 #include <memory>
 #include <unordered_map>
+#include <x86intrin.h>
 
 #include "core/inc/driver.h"
 #include "core/inc/memory_region.h"
+#include "core/driver/xdna/uapi/amdxdna_accel.h"
+
+// Flushes the cache lines assocaited with a BO. This
+// is used to sync a BO without going to the xdna driver.
+// This is from the XRT KMQ shim.
+inline void
+clflush_data(const void *base, size_t offset, size_t len)
+{
+  static long cacheline_size = 0;
```

Comment:
I don't think this needs to be static.

Reply:
Good point. Will remove.

Reply:
Would you prefer that over just leaving the variable as static? Either works for me.

Reply:
Well if it is a member of the driver, then it should just be something that is initialized upon construction or cached on first call. However, I don't think this belongs in the XDNA driver and should be made a util function and hence a static variable defined in the util lib.

Reply:
BTW, in C++17 it's this https://en.cppreference.com/w/cpp/thread/hardware_destructive_interference_size